TOOLS/odoo-tools.sh

```diff
@@ -1224,23 +1227,72 @@ if [ "$SCRIPT_MODE" = "updatetranslation" ]; then
     echo -e "--------------------------------------------------------------------------------------------------------"
-    if [ $# -ne 2 ]; then
-        echo -e "ERROR: \"setup-toosl.sh $SCRIPT_MODE\" takes exactly four arguments!"
+    if ! [ $# -ge 3 ]; then
+        echo -e "ERROR: \"setup-toosl.sh $SCRIPT_MODE\" takes exactly three arguments!"
         exit 2
     fi
-
-    # TODO: check or create INSTANCE/DATABASE/BACKUP folder
-    # Todo: CD to BACKUP folder
-    # Todo: create backup with dbname--DATE.zip with DATE in Format 2014-12-30--24-59
-
-
+    TYPE=$4
+    DBNAME=$3
+    TARGET_BRANCH=$2
+    INSTANCE_PATH="${REPOPATH}/${TARGET_BRANCH}"
+    #if [ "x${TYPE}" == "x" ]; then
+    #    TYPE="zip"
+    #fi
+    echo "INSTANCE_PATH --> ${INSTANCE_PATH}"
+    # Todo: check vmware Snapshot how to remote execute the vmware-cmd command with ssh connection to esx server directly, check if the VM is running on this machine
+    # Todo: or find a way of acting through Virtual center server this server has access to the whole cluster and no check on which host a machine is running would be needed
+    # Check if a database with this name exists
+    if [ `su - postgres -c "psql -l | grep ${DBNAME} | wc -l"` -gt 0 ]; then
+        echo -e "Database ${DBNAME} exists, starting to backup this datase ... "
+    elif [ ${DBNAME} = "all" ]; then
+        echo -e "All databases going to be backed up...."
+    elif [ ${DBNAME} = "all_running" ]; then
+        echo -e "All RUNNING databases going to be backed up...."
+    else
+        echo -e "check your Databasename, you gave ${DBNAME}, but this seems not to exist, stopping script......"
+        exit 2
+    fi
+    if [ ${DBNAME} = "all_running" ]; then
+        DATABASE_RUNNING=($(ps -ef|grep "openerp-server*" |awk '{printf $13;printf "\n"; }')) #TODO: check aber auch ALLE Prostgres Prozesse
+        for i in "${DATABASE_RUNNING[@]}"; do
+            #store running databases and log do
+            #getting config of database
+            DATABASECONFIGFILE=${INSTANCE_PATH}/${i}/${i}.conf
+            echo "configflepath --> ${DATABASECONFIGFILE}"
+            BASEPORT69=($(grep "xmlrpc_port" ${DATABASECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+            SUPER_PASSWORD=($(grep "admin_passwd" ${DATABASECONFIGFILE} | awk '{printf $3;printf "\n"; }'))
+            BACKUPFILENAME=${INSTANCE_PATH}/${i}/BACKUP/IS-BACKUP--${i}--`date +%Y-%m-%d__%H-%M`.zip
+            echo "backup all Databases, while now backing up ${i} ...."
+            echo -e $(${INSTANCE_PATH}/TOOLS/db-tools.py -b ${BASEPORT69} -s ${SUPER_PASSWORD} "backup" -d ${i} -f ${BACKUPFILENAME}) #-t ${TYPE})
+        done
+    elif [ ${DBNAME} = "all" ]; then
+        DATABASE_RUNNING=($(su - postgres -c "psql --tuples-only -P format=unaligned -c \"SELECT datname FROM pg_database WHERE NOT datistemplate AND datname <> 'postgres'\"|grep -v -e _cloud -e _pad")) #TODO: check aber auch ALLE Prostgres Prozesse
```

Comment:
Varaible Name is missleading: DATABASE_RUNNING (we do not know if these are up and running odoo instances)
DATABASES would work here fine